tools/build_api.py

```diff
@@ -250,6 +251,15 @@ def find_valid_toolchain(target, toolchain):
                 and "uARM" in {toolchain_name, target.default_toolchain}
             ):
                 end_warnings.append(UARM_TOOLCHAIN_WARNING)
+
+            if is_std_c_lib_supported_instead_small_c_lib(target, toolchain):
+                warning = (
+                    "Warning: We noticed that you are using target.c_lib set to small.\n"
+                    "As the {} target does not support a small C library for the {} toolchain,\n"
+                    "we are using the standard C library instead. "
+                ).format(target.name, toolchain)
+                end_warnings.append(warning)
+
```

Comment:
Since the library choice is likely to be set by the target this might come as a surprise (it may well not be the user that sets 'small'). Consider rewording...

Reply:
Please review 0e9c83e15985570a1b18cea3e794d8f92c9d9ee1